### ISSUE TITLE
Tox source testing errors, failing gh action

### ISSUE THREAD

**Issue by cpstrommen** [LibreCS/vqgarm#1]
On current execution of tox buld test gh action, the out error code is as follows:
```bash
Run python -m tox -e py
.package create: /home/runner/work/vqgarm/vqgarm/.tox/.package
.package installdeps: hatchling
py create: /home/runner/work/vqgarm/vqgarm/.tox/py
py installdeps: flake8
py inst: /home/runner/work/vqgarm/vqgarm/.tox/.tmp/package/1/vqgarm-0.0.1.tar.gz
py installed: flake8==4.0.1,mccabe==0.6.1,pycodestyle==2.8.0,pyflakes==2.4.0,vqgarm @ file:///home/runner/work/vqgarm/vqgarm/.tox/.tmp/package/1/vqgarm-0.0.1.tar.gz
py run-test-pre: PYTHONHASHSEED='26309[7](https://github.com/LibreCS/vqgarm/runs/7583531261?check_suite_focus=true#step:5:8)4920'
py run-test: commands[0] | flake[8](https://github.com/LibreCS/vqgarm/runs/7583531261?check_suite_focus=true#step:5:9) .
./predict.py:146:34: F821 undefined name 'urlopen'
./predict.py:315:17: F541 f-string is missing placeholders
./predict.py:323:17: F541 f-string is missing placeholders
./predict.py:672:44: F821 undefined name 'z_orig'
./predict.py:673:13: W503 line break before binary operator
./predict.py:674:13: W503 line break before binary operator
./predict.py:701:[9](https://github.com/LibreCS/vqgarm/runs/7583531261?check_suite_focus=true#step:5:10): W503 line break before binary operator
./predict.py:702:9: W503 line break before binary operator
./predict.py:711:46: E203 whitespace before ':'
./predict.py:729:13: W503 line break before binary operator
./generate.py:72:90: F541 f-string is missing placeholders
./generate.py:73:98: F541 f-string is missing placeholders
./generate.py:154:17: W503 line break before binary operator
./generate.py:628:35: E127 continuation line over-indented for visual indent
./generate.py:964:20: E128 continuation line under-indented for visual indent
./generate.py:[10](https://github.com/LibreCS/vqgarm/runs/7583531261?check_suite_focus=true#step:5:11)06:58: W[29](https://github.com/LibreCS/vqgarm/runs/7583531261?check_suite_focus=true#step:5:30)2 no newline at end of file
ERROR: InvocationError for command /home/runner/work/vqgarm/vqgarm/.tox/py/bin/flake8 . (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py: commands failed
Error: Process completed with exit code 1.
```
Most fixes are minor and inconsequential in nature, tox runs on commit so feel free to make iterative changes

**Comment by cpstrommen** [comment]
Fixed a few out of scope variable calls and flake8 ignored codes F541 and W503 and now the test errors out to:
```bash
Run python -m tox -e py
.package create: /home/runner/work/vqgarm/vqgarm/.tox/.package
.package installdeps: hatchling
py create: /home/runner/work/vqgarm/vqgarm/.tox/py
py installdeps: flake8
py inst: /home/runner/work/vqgarm/vqgarm/.tox/.tmp/package/1/vqgarm-0.0.1.tar.gz
py installed: flake8==4.0.1,mccabe==0.6.1,pycodestyle==2.8.0,pyflakes==2.4.0,vqgarm @ file:///home/runner/work/vqgarm/vqgarm/.tox/.tmp/package/1/vqgarm-0.0.1.tar.gz
py run-test-pre: PYTHONHASHSEED='474396708'
py run-test: commands[0] | flake8 .
./predict.py:712:46: E203 whitespace before ':'
./generate.py:628:35: E127 continuation line over-indented for visual indent
./generate.py:964:20: E128 continuation line under-indented for visual indent
./generate.py:1006:58: W292 no newline at end of file
ERROR: InvocationError for command /home/runner/work/vqgarm/vqgarm/.tox/py/bin/flake8 . (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py: commands failed
Error: Process completed with exit code 1.
```
More dumb whitespace fixes, will inform if structural or function changes are needed.

**Comment by cpstrommen** [comment]
Fixed formatting issues but E128 issue on ln964 caused a syntax error:
```bash
.package create: /home/runner/work/vqgarm/vqgarm/.tox/.package
.package installdeps: hatchling
py create: /home/runner/work/vqgarm/vqgarm/.tox/py
py installdeps: flake[8](https://github.com/LibreCS/vqgarm/runs/7584767490?check_suite_focus=true#step:5:9)
py inst: /home/runner/work/vqgarm/vqgarm/.tox/.tmp/package/1/vqgarm-0.0.1.tar.gz
py installed: flake8==4.0.1,importlib-metadata==4.2.0,mccabe==0.6.1,pycodestyle==2.8.0,pyflakes==2.4.0,typing_extensions==4.3.0,vqgarm @ file:///home/runner/work/vqgarm/vqgarm/.tox/.tmp/package/1/vqgarm-0.0.1.tar.gz,zipp==3.8.1
py run-test-pre: PYTHONHASHSEED='28[9](https://github.com/LibreCS/vqgarm/runs/7584767490?check_suite_focus=true#step:5:10)56086[10](https://github.com/LibreCS/vqgarm/runs/7584767490?check_suite_focus=true#step:5:11)'
py run-test: commands[0] | flake8 .
./generate.py:964:[18](https://github.com/LibreCS/vqgarm/runs/7584767490?check_suite_focus=true#step:5:19): E999 SyntaxError: invalid syntax
ERROR: InvocationError for command /home/runner/work/vqgarm/vqgarm/.tox/py/bin/flake8 . (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py: commands failed
Error: Process completed with exit code 1.
```
Plan now is to revert change on ln964 and add flake8 ignore for E128. Will inform on future issues

**Comment by cpstrommen** [comment]
Fixed with flake8 ignore tags as of: [commit d0bd331](https://github.com/LibreCS/vqgarm/commit/d0bd331c30d73f0827298d92c779d9b39ee3aa8b)
In the future, we fix problems directly not with ignore tags.
